e2e/src/test/java/no/sysco/testing/kafka/e2e/ContainersIT.java

```diff
@@ -0,0 +1,194 @@
+
+                            assertNotNull(jsonPresentation);
+                        });
+    }
+
+    private static void setZookeeperAndKafka() {
+        kafka = new KafkaContainer(CONFLUENT_PLATFORM_VERSION);
+        kafka.start();
+    }
+
+    private static void setSchemaRegistry() {
+        // get network alias to be able connect other container to env
+        KAFKA_BROKER_INSIDE_DOCKER_ENV = "PLAINTEXT://" + kafka.getNetworkAliases().get(0) + ":9092";
+        schemaRegistry =
+                new GenericContainer("confluentinc/cp-schema-registry:" + CONFLUENT_PLATFORM_VERSION)
+                        .withExposedPorts(8081)
+                        .withNetwork(kafka.getNetwork())
+                        .withEnv("SCHEMA_REGISTRY_HOST_NAME", "localhost")
+                        .withEnv("SCHEMA_REGISTRY_LISTENERS", "http://0.0.0.0:8081")
+                        .withEnv("SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS", KAFKA_BROKER_INSIDE_DOCKER_ENV)
+                        .waitingFor(Wait.forHttp("/subjects").forStatusCode(200));
+        schemaRegistry.start();
+        SCHEMA_REGISTRY_INSIDE_DOCKER_ENV =
+                "http://" + schemaRegistry.getNetworkAliases().get(0) + ":8081";
+    }
+
+    private static void setJsonServer() {
+        jsonServer =
+                new GenericContainer("zhenik/json-server")
+                        .withExposedPorts(80)
+                        // all containers put in same network
+                        .withNetwork(kafka.getNetwork())
+                        .withEnv("ID_MAP", "id")
+                        .withNetworkAliases("json-server")
+                        .withClasspathResourceMapping(
+                                "json-server-database.json", "/data/db.json", BindMode.READ_WRITE)
+                        .waitingFor(Wait.forHttp("/").forStatusCode(200));
+
+        jsonServer.start();
+        // provide availability make http calls from localhost against docker env
+        JSON_SERVER_EXPOSED = LOCAL_HOST + ":" + jsonServer.getMappedPort(80);
+        JSON_SERVER_INSIDE_DOCKER_ENV = "http://" + jsonServer.getNetworkAliases().get(0) + ":80";
+    }
+
+    private static void setHttProducer() {
+        httpProducer =
+                new GenericContainer("zhenik/http-producer:data-pipeline")
+                        .withExposedPorts(8080)
+                        .withNetwork(kafka.getNetwork())
+                        .withEnv("APPLICATION_PORT", "8080")
+                        .withEnv("KAFKA_BOOTSTRAP_SERVERS", KAFKA_BROKER_INSIDE_DOCKER_ENV)
+                        .withEnv("SCHEMA_REGISTRY_URL", SCHEMA_REGISTRY_INSIDE_DOCKER_ENV)
+                        .withEnv("SINK_TOPIC", TOPIC)
+                        .waitingFor(Wait.forHttp("/messages").forStatusCode(200));
+        httpProducer.start();
+
+        HTTP_PRODUCER_EXPOSED = LOCAL_HOST + ":" + httpProducer.getMappedPort(8080);
```

Comment:
please always use `getContainerIpAddress` instead of `localhost`, see https://www.testcontainers.org/features/networking/#getting-the-container-ip-address

Reply:
done